src/Paramore.Brighter.Extensions.DependencyInjection/ServiceProviderHandlerFactory.cs

```diff
@@ -43,49 +45,76 @@ public ServiceProviderHandlerFactory(IServiceProvider serviceProvider)
         {
             _serviceProvider = serviceProvider;
             var options = (IBrighterOptions) serviceProvider.GetService(typeof(IBrighterOptions));
-            if (options == null) _isTransient = true; else _isTransient = options.HandlerLifetime == ServiceLifetime.Transient;
+            if (options == null) _isSingleton = false; else _isSingleton = options.HandlerLifetime == ServiceLifetime.Singleton;
         }
 
         /// <summary>
         /// Creates an instance of the request handler
         /// Lifetime is set during registration
         /// </summary>
         /// <param name="handlerType">The type of handler to request</param>
+        /// <param name="lifetime">The brighter Handler lifetime</param>
         /// <returns>An instantiated request handler</returns>
-        IHandleRequests IAmAHandlerFactorySync.Create(Type handlerType)
+        IHandleRequests IAmAHandlerFactorySync.Create(Type handlerType, IAmALifetime lifetime)
         {
-            return (IHandleRequests)_serviceProvider.GetService(handlerType);
+            if (_isSingleton)
+                return (IHandleRequests)_serviceProvider.GetService(handlerType);
+
+            if (!_scopes.ContainsKey(lifetime))
+                _scopes.Add(lifetime, _serviceProvider.CreateScope());
+            
+            return (IHandleRequests)_scopes[lifetime].ServiceProvider.GetService(handlerType);
         }
 
         /// <summary>
         /// Creates an instance of the request handler
         /// Lifetime is set during registration
         /// </summary>
         /// <param name="handlerType">The type of handler to request</param>
+        /// <param name="lifetime">The brighter Handler lifetime</param>
         /// <returns>An instantiated request handler</returns>
-        IHandleRequestsAsync IAmAHandlerFactoryAsync.Create(Type handlerType)
+        IHandleRequestsAsync IAmAHandlerFactoryAsync.Create(Type handlerType, IAmALifetime lifetime)
         {
-            return (IHandleRequestsAsync)_serviceProvider.GetService(handlerType);
+            if (_isSingleton)
+                return (IHandleRequestsAsync)_serviceProvider.GetService(handlerType);
+            
```

Comment:
Agreed that is this handles our factory implementation for ServiceCollection, it is a good place to abstract how that works